### PR TITLE
initial implementation of google auth

### PR DESCRIPTION
**Pull request by szuecs**
#17 initial Google OAuth2 integration.

### CODE REVIEW

File: google/google.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+	"golang.org/x/oauth2/google"
+)
+
+// Credentials stores google client-ids.
+type Credentials struct {
+	ClientID     string `json:"clientid"`
+	ClientSecret string `json:"secret"`
+}
+
+// User is a retrieved and authentiacted user.
```
Comment (Raffo) [inline google/google.go]: typo authentiacted -> authenticated

File: google/google.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+	return base64.StdEncoding.EncodeToString(b)
+}
+
+// Setup the authorization path
+func Setup(redirectURL, credFile string, scopes []string, secret []byte) {
+	store = sessions.NewCookieStore(secret)
+	var c Credentials
+	file, err := ioutil.ReadFile(credFile)
+	if err != nil {
+		fmt.Printf("File error: %v\n", err)
```
Comment (Raffo) [inline google/google.go]: Do you really wanted to use fmt?

File: google/google.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+	session.Set("state", state)
+	session.Save()
+	ctx.Writer.Write([]byte("<html><title>Golang Google</title> <body> <a href='" + GetLoginURL(state) + "'><button>Login with Google!</button> </a> </body></html>"))
+}
+
+func GetLoginURL(state string) string {
+	return conf.AuthCodeURL(state)
+}
+
+// Auth is the google authorziation middleware. You can use them to protect a routergroup.
```
Comment (Raffo) [inline google/google.go]: Do you want to have this example here as commented code?

Comment (szuecs) [inline google/google.go]: yes it's for godoc

**Comment by Raffo** [comment]
LGTM, but travis is not happy, maybe we should address this in a following issue/pr.

**Pull request merged**